rest_api/api/auth.py

```diff
@@ -0,0 +1,16 @@
+from rest_framework.authentication import BasicAuthentication
+from django.contrib.auth.models import User
+from rest_framework.exceptions import AuthenticationFailed
+
+class CustomAuthentication(BasicAuthentication):
+    def authenticate(self, request):
+        username = request.GET.get('username')
+        if username is None:
+            return None
```

Comment:
why return None ???? return a message that please provide username

Reply:
Done